### ISSUE TITLE
Web adapter: IndexedDB persistence

### ISSUE THREAD

**Issue by schickling** [livestorejs/livestore#131]
Currently the web adapter only supports persisting to OPFS. It could be useful for some scenarios to also allow persisting to IndexedDB.

Note: In older LiveStore versions there used to be preliminary support for IndexedDB but it was very rudimentary and not production-ready.

**Comment by joodaloop** [comment]
Will this be similar to [AbsurdSQL](https://github.com/jlongster/absurd-sql)?

**Comment by schickling** [comment]
Yes, LiveStore is using a fork of [wa-sqlite](https://github.com/livestorejs/wa-sqlite) WASM build, which requires a dedicated VFS layer, which we currently have a implementation targeting OPFS. Which is running on a dedicated worker using the synchronous OPFS API. If we wanted to also support IndexedDB as a VFS target, we would need to introduce some changes in the WA SQLite fork in order to also support asynchronous APIs, which IndexedDB requires.
